desc/objectives/_bootstrap.py

```diff
@@ -125,6 +133,18 @@ def build(self, use_jit=True, verbose=1):
             ValueError,
             "Helicity toroidal mode number should be 0 (QA) or +/- NFP (QH)",
         )
+        rho = grid.nodes[grid.unique_rho_idx, 0]
+        errorif(
+            np.any(np.isclose(rho, 0)),
+            ValueError,
+            "Redl formula is undefined" "at rho=0, but grid has grid points at rho=0",
+        )
+
+        errorif(
+            np.any(np.isclose(rho, 1)),
+            ValueError,
+            "Redl formula is undefined" "at rho=1, but grid has grid points at rho=1",
+        )
```

Comment:
The Redl formula is defined at $\rho=1$, but not where $p=0$ which is often the case at the boundary. We should change this to check if $n$ or $T$ vanish at any of the grids points. Or at the very least change this from an error to a warning.